File: 1657-determine-if-two-strings-are-close/close_strings.py
def close_strings(word1, word2):
    if len(word1) != len(word2):
        return False

    freq_map1 = get_character_frequency_map(word1)
    freq_map2 = get_character_frequency_map(word2)

    chars1 = ''.join(sorted(freq_map1.keys()))
    chars2 = ''.join(sorted(freq_map2.keys()))

    freqs1 = ''.join(map(str, sorted(freq_map1.values())))
    freqs2 = ''.join(map(str, sorted(freq_map2.values())))

    return chars1 == chars2 and freqs1 == freqs2

def get_character_frequency_map(word):
    freq_map = {}
    for char in word:
        freq_map[char] = freq_map.get(char, 0) + 1
    return freq_map

# Example usage:
word1 = "cabbba"
word2 = "abbccc"
result = close_strings(word1, word2)
print(result)  # Output: True

File: 1657-determine-if-two-strings-are-close/close_strings.js
function closeStrings(word1, word2) {
    if (word1.length !== word2.length) {
        return false;
    }

    const freqMap1 = getCharacterFrequencyMap(word1);
    const freqMap2 = getCharacterFrequencyMap(word2);

    const chars1 = Array.from(freqMap1.keys()).sort().join('');
    const chars2 = Array.from(freqMap2.keys()).sort().join('');

    const freqs1 = Array.from(freqMap1.values()).sort().join('');
    const freqs2 = Array.from(freqMap2.values()).sort().join('');

    return chars1 === chars2 && freqs1 === freqs2;
}

function getCharacterFrequencyMap(word) {
    const freqMap = new Map();
    for (const char of word) {
        freqMap.set(char, (freqMap.get(char) || 0) + 1);
    }
    return freqMap;
}

// Example usage:
const word1 = "cabbba";
const word2 = "abbccc";
const result = closeStrings(word1, word2);
console.log(result); // Output: true
